lib/Conversion/TritonGPUToLLVM/Utility.cpp

```diff
@@ -231,12 +231,30 @@ emitIndicesUsingLinearLayouts(Location loc, RewriterBase &rewriter,
       withCTAOffset ? target.getClusterCTAId(rewriter, loc) : i32_val(0);
   unsigned rank = shape.size();
   SmallVector<SmallVector<Value>> ret;
+  // Linear layout function is split in two parts below:
+  // L(r, t, w, b) = L(0, t, w, b) xor L(r, 0, 0, 0)
+  //     idxs      =    idxsBase   xor    idxsReg
+  //
+  // L(0, t, w, b) part is same for all registers, nested out of reg loop
```

Comment:
Nit: ".. is the same for all registers, so we hoist it out of the main register loop in the below."